src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.ts

```diff
@@ -40,10 +39,13 @@ export class PipelinesWidgetComponent implements OnInit {
 
   ngOnInit() {
     this.contextPath = this.context.current.map(context => context.path);
-    this.buildConfigs = this.pipelinesService.current
-      .publish();
-    this.buildConfigsCount = this.buildConfigs.map(buildConfigs => buildConfigs.length);
-    this.buildConfigs.connect();
+    this.buildConfigs = this.pipelinesService.current.share();
+    // buildConfigsCount triggers changes in the DOM; force Angular Change Detection
+    // via setTimeout encapsulation
+    this.buildConfigs
+      .map(buildConfigs => buildConfigs.length)
+      .share()
+      .subscribe(length => setTimeout(() => this.buildConfigsCount = length));
```

Comment:
Did we figure out how the ConnectableObservable was working around the change detection? This makes sense but I'm not clear on how/why it was working before...

Reply:
I can only surmise it's purely a coincidence with the timing. Removing the Connectable publish/connect, I can refresh the page multiple times and, in the majority see a CD error and a minority, less than 1/10, not see it.